src/braket/aws/aws_device.py

```diff
@@ -622,3 +666,133 @@ def get_device_region(device_arn: str) -> str:
             )
+
+    def refresh_gate_calibrations(self) -> GateCalibrations:
+        """
+        Refreshes the gate calibration data upon request.
+
+        If the device does not have calibration data, None is returned.
+
+        Returns:
+            GateCalibrations: the calibration data for the device.
+        """
+        if self.gate_calibrations_href is not None:
+            try:
+                with urllib.request.urlopen(self.gate_calibrations_href.split("?")[0]) as f:
+                    json_calibration_data = self._parse_calibration_json(
+                        json.loads(f.read().decode("utf-8"))
+                    )
+                    return GateCalibrations(json_calibration_data)
+            except urllib.error.URLError:
+                raise urllib.error.URLError(f"Unable to reach {self.gate_calibrations_href}")
+        else:
+            return None
+
+    def _get_waveforms(self, waveforms_json: str) -> Dict:
+        waveforms = dict()
+        for waveform in waveforms_json:
+            w = waveforms_json[waveform]
+            if "amplitudes" in w.keys():
+                parsed_waveform = ArbitraryWaveform._from_json(w)
+                waveforms[parsed_waveform.id] = parsed_waveform
+            elif w["name"] == "drag_gaussian":
+                parsed_waveform = DragGaussianWaveform._from_json(w)
+                waveforms[parsed_waveform.id] = parsed_waveform
+            elif w["name"] == "gaussian":
+                parsed_waveform = GaussianWaveform._from_json(w)
+                waveforms[parsed_waveform.id] = parsed_waveform
+            elif w["name"] == "constant":
+                parsed_waveform = ConstantWaveform._from_json(w)
+                waveforms[parsed_waveform.id] = parsed_waveform
+            else:
+                id = w["waveformId"]
+                raise ValueError(f"The waveform {id} of cannot be constructed")
+        return waveforms
+
+    def _get_pulse_sequence(
+        self, calibration: Dict, waveforms: Dict[ArbitraryWaveform]
+    ) -> PulseSequence:
+        calibration_sequence = PulseSequence()
+        for instruction in range(len(calibration)):
+            instr = calibration[instruction]
+            if hasattr(calibration_sequence, f"_parse_{instr['name']}_json"):
+                instr_parser = getattr(calibration_sequence, f"_parse_{instr['name']}_json")
+                instr_function_args = (
+                    instr_parser(instr, self.frames)
+                    if instr["name"] != "play"
+                    else instr_parser(instr, waveforms, self.frames)
+                )
+                instr_function = getattr(calibration_sequence, instr["name"])
+                calibration_sequence = (
+                    instr_function(instr_function_args)
+                    if len(instr_function_args) <= 1
+                    else instr_function(*instr_function_args)
+                )
```

Comment:
- Why is this logic in the aws_device class?
- Why are we relying on method names following a specific format which isn't enforced by an interface?

Reply:
moved the logic